doc/developer/platform/architecture-storage.md

```diff
@@ -0,0 +1,131 @@
+## Storage architecture
+
+STORAGE is responsible for creating and maintaining definite time varying
+collections (TVCs) that contain either ingested data from an external source
+(e.g Kafka), data sinked from one or more CLUSTER instances, or data coming
+from `Insert` requests to the storage controller, providing a table-like
+experience.
+
+## The STORAGE collection structure
+
+![https://imgur.com/dzB8kid.png](https://imgur.com/dzB8kid.png)
+
+### Overview
+
+STORAGE is tasked with managing the lifecycle of STORAGE collections (just
+collections hereafter). Collections support concurrent, sharded writes from
+multiple CLUSTER instances configured in HA mode. Furthermore, collections
```

Comment:
Active Replication mode? HA seems to be more the concept that matters to the clients, and AR is the mechanism.

Reply:
Yeap, this was written before we decided to split the terms.